python/hsfs/core/deltastreamer_jobconf.py

```diff
@@ -18,17 +18,18 @@
 import json
 
 from hsfs import util
+from hsfs.core.job_configuration import JobConfiguration
 
 
 class DeltaStreamerJobConf:
     def __init__(self, options, spark_options, **kwargs):
         self._options = options
-        self._spark_options = spark_options
+        self._spark_options = JobConfiguration(**spark_options if spark_options else {}).to_dict()
```

Comment:
I'm still confused by this operation - The spark options should already be a JobConfiguration based on the documentation: https://docs.hopsworks.ai/hopsworks-api/latest/generated/api/feature_group_api/#insert

Maybe we can start accepting a dictionary as well - but right now if a user follows the documentation, this code will break their workload.

Reply:
now it supports both `dict` and `JobConfiguration`